sw/device/sca/aes_serial/aes_serial.c

```diff
@@ -0,0 +1,334 @@
+ */
+static simple_serial_result_t simple_serial_set_key(const aes_cfg_t *aes_cfg,
+                                                    const char *key_share0,
+                                                    size_t key_len) {
+  // The implementation does not use key shares to simplify AES attacks.
+  static const uint8_t kKeyShare1[32] = {0};
+
+  if (key_len != kAesKeySize) {
+    return kSimpleSerialError;
+  }
+
+  aes_clear();
+  while (!aes_idle()) {
+    ;
+  }
+  aes_init(*aes_cfg);
+
+  aes_key_put(key_share0, kKeyShare1, aes_cfg->key_len);
+  return kSimpleSerialOk;
+}
+
+/*
+ * Simple serial trigger AES encryption command.
+ *
+ * @param aes_cfg        AES configuration object.
+ * @param plain_text     plain text to be encrypted.
+ * @param plain_text_len plain text length.
+ *
+ * @return kSimpleSerialOk on success, kSimpleSerialError otherwise.
+ */
+static simple_serial_result_t simple_serial_trigger_encryption(
+    const char *plain_text, size_t plain_text_len) {
+  if (plain_text_len > kMaxInputLengthBin) {
+    return kSimpleSerialError;
+  }
+
+  char cipher_text[64];
+  size_t num_blocks = get_num_blocks(plain_text_len);
+  if (num_blocks != 1) {
+    return kSimpleSerialError;
+  }
+
+  // start timer to wake up Ibex after AES is done.
+  uint64_t current_time;
+  CHECK(dif_rv_timer_counter_read(&timer, kHart, &current_time) ==
+        kDifRvTimerOk);
+  CHECK(dif_rv_timer_arm(&timer, kHart, kComparator,
+                         current_time + kDeadline) == kDifRvTimerOk);
+  CHECK(dif_rv_timer_counter_set_enabled(&timer, kHart, kDifRvTimerEnabled) ==
+       kDifRvTimerOk);
+
+  aes_data_put_wait(plain_text);
+  if (dif_gpio_all_write(&gpio, kGpioCaptureTriggerH) != kDifGpioOk) {
+    return kSimpleSerialError;
+  }
+
+  aes_manual_trigger();
+  wait_for_interrupt();
+
+  if (dif_gpio_all_write(&gpio, kGpioCaptureTriggerL) != kDifGpioOk) {
+    return kSimpleSerialError;
+  }
+  aes_data_get_wait(cipher_text);
+
+  print_cmd_response('r', cipher_text, plain_text_len);
+  return kSimpleSerialOk;
+}
+
+/**
+ * Handle simple serial command.
+ *
+ * @param aes_cfg AES configuration object.
+ * @param cmd     input command buffer.
+ * @param cmd_len number of characters set in input buffer.
+ */
+static void simple_serial_handle_command(const aes_cfg_t *aes_cfg,
+                                         const char *cmd, size_t cmd_len) {
+  // Data length is half the size of the hex encoded string.
+  const size_t data_len = cmd_len >> 1;
+  if (data_len >= kMaxInputLengthBin) {
+    return;
+  }
+  char data[64] = {0};
+
+  // The simple serial protocol does not expect return status codes for invalid
+  // data, thus it is ok to return early.
+  if (a2b_hex(&cmd[1], data, data_len) == kSimpleSerialError) {
+    return;
+  }
+
+  if (cmd[0] == 'v') {
+    print_cmd_response('z', (const char *)&kSimpleSerialProtocolVersion,
+                       /*data_len=*/1);
+    return;
+  }
```

Comment:
Is this intended? Asking because this is also checked in the `switch` below.

Reply:
Removed `v` from the `switch` below.